src/main/java/org/sefglobal/scholarx/service/MenteeService.java

```diff
@@ -127,21 +135,38 @@ public Mentee updateAssignedMentor(long menteeId, Long mentorId)
             log.error(msg);
             throw new ResourceNotFoundException(msg);
         }
-        
+
+        Mentor previouslyAssignedMentor = optionalMentee.get().getAssignedMentor();
+        int menteeCountCurrentMentor = optionalMentor.get().getNoOfAssignedMentees();
+        int menteeCountPreviousMentor = 0;
+        if (previouslyAssignedMentor != null) {
+            menteeCountPreviousMentor = previouslyAssignedMentor.getNoOfAssignedMentees();
+        }
         ProgramState programState = optionalMentee.get().getProgram().getState();
         if (programState.equals(ProgramState.ADMIN_MENTEE_FILTRATION)) {
             optionalMentee.get().setAssignedMentor(optionalMentor.get());
             optionalMentee.get().setState(EnrolmentState.ASSIGNED);
+            menteeCountCurrentMentor++;
+            menteeCountPreviousMentor--;
         } else if (programState.equals(ProgramState.WILDCARD)) {
             optionalMentee.get().setAssignedMentor(optionalMentor.get());
             optionalMentee.get().setState(EnrolmentState.APPROVED);
+            menteeCountCurrentMentor++;
+            menteeCountPreviousMentor--;
         } else {
             String msg = "Error, Mentee cannot be updated. " +
                     "Program is not in a valid state.";
             log.error(msg);
             throw new BadRequestException(msg);
         }
-        
+
+        if (previouslyAssignedMentor != null) {
+            previouslyAssignedMentor.setNoOfAssignedMentees(menteeCountPreviousMentor);
+            mentorRepository.save(previouslyAssignedMentor);
+        }
+
+        optionalMentor.get().setNoOfAssignedMentees(menteeCountCurrentMentor);
+        optionalMentee.get().setAssignedMentor(optionalMentor.get());
```

Comment:
I think we can remove lines 147 and 152 because it sets the assigned mentor here again.
(`optionalMentee.get().setAssignedMentor(optionalMentor.get());`)